.circleci/config.yml

```diff
@@ -0,0 +1,115 @@
+# These environment variables must be set in CircleCI UI
+#
+# NPM_TOKEN - A valid NPM token for releases
+version: 2.1
+
+defaults: &defaults
+  working_directory: ~/addons-scanner-utils
+  docker:
+    # This is the NodeJS version we run in production.
+    - image: circleci/node:10
+
+defaults-next: &defaults-next
+  working_directory: ~/addons-scanner-utils
+  docker:
+    # This is the NodeJS version we run in production.
+    - image: circleci/node:12
+
+jobs:
+  test:
+    <<: *defaults
+    steps:
+      - checkout
+      - restore_cache:
+          name: restore yarn package cache
+          keys:
+            - yarn-packages-{{ checksum "yarn.lock" }}
+      - run:
+          name: install dependencies
+          # See: https://github.com/mozilla/addons-frontend/issues/3034
+          command: yarn install --pure-lockfile
+      - save_cache:
+          name: save yarn package cache
+          key: yarn-packages-{{ checksum "yarn.lock" }}
+          paths:
+            - ~/.cache/yarn
+      - run:
+          name: unit tests
+          command: yarn test-ci
+      - run:
+          # Make sure we can build a production package.
+          name: package module
+          command: yarn pack
+      - run:
+          name: type checking
+          command: yarn typecheck
+      - run:
+          name: type coverage
+          command: yarn type-coverage
+      - run:
+          name: prettier
+          command: yarn prettier-ci
+      - run:
+          name: linter
+          command: yarn lint
+      # Set up a workspace to share data between this job and the `release-tag`
+      # one when we want to publish a new npm version.
+      - persist_to_workspace:
+          root: ~/addons-scanner-utils
+          paths: .
+      - store_artifacts:
+          path: coverage
+
+  test-next:
+    <<: *defaults-next
+    steps:
+      - checkout
+      - restore_cache:
+          name: restore yarn package cache
+          keys:
+            - next-yarn-packages-{{ checksum "yarn.lock" }}
+      - run:
+          name: install dependencies
+          # See: https://github.com/mozilla/addons-frontend/issues/3034
+          command: yarn install --pure-lockfile
+      - save_cache:
+          name: save yarn package cache
+          key: next-yarn-packages-{{ checksum "yarn.lock" }}
+          paths:
+            - ~/.cache/yarn
+      - run:
+          name: unit tests
+          command: yarn test-ci
+
+  release-tag:
+    <<: *defaults
+    steps:
+      - attach_workspace:
+          at: ~/addons-scanner-utils
+      - run:
+          name: authenticate with registry
+          command: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/repo/.npmrc
+      - run:
+          name: publish package
+          command: npm publish
+
+workflows:
+  version: 2
+  default-workflow:
+    jobs:
+      - test:
+          filters: # required since `release-tag` has tag filters AND requires `test`
```

Comment:
Comment is appreciated since lack of familiarity makes understanding this config much harder.